File: 531_LonelyPixel1/main.java
class Solution {
    public int findLonelyPixel(char[][] picture) {
        int m = picture.length;
        if(m == 0) return 0;
        int n = picture[0].length;
        if(n == 0) return 0;
        
        int[] rows = new int[m];
        int[] cols = new int[n];
        
        for(int i = 0; i < m; i++){
            for(int j = 0; j < n; j++){
                rows[i] += picture[i][j] == 'B' ? 1 : 0;
                cols[j] += picture[i][j] == 'B' ? 1 : 0;
            }
        }
        
        int lonely = 0;
        for (int i = 0; i < m; i++) {
            for (int j = 0; j < n && rows[i] > 0; j++) {
                lonely += (picture[i][j] == 'B' && rows[i] == 1 && cols[j] == 1) ? 1 : 0;
            }
        }

        return lonely;     
    }
}

File: 531_LonelyPixel1/main.cpp
class Solution{
public:
    int findLonelyPixel(vector<vector<char>>& picture){
        int m = picture.size();
        if(m == 0) return 0;
        int n = picture[0].size();
        if(n == 0) return 0;
        
        vector<int> rows(m, 0);
        vector<int> cols(n, 0);
        
        for(int i = 0; i < m; i++){
            for(int j = 0; j < n; j++){
                rows[i] += picture[i][j] == 'B';
                cols[i] += picture[i][j] == 'B';
            }
        }
        
        int ans = 0;
        for(int i = 0; i < m; i++){
            for(int j = 0; j < n; j++){
                ans += picture[i][j] == 'B' && rows[i] == 1 && cols[j] == 1;
            }
        }
        return ans;
    }
};
